kore/src/Kore/Internal/Predicate.hs

```diff
@@ -119,438 +184,642 @@ import Pretty
+fromPredicate sort = Recursive.fold worker
+  where
+    worker (pat :< predF) = TermLike.setSimplified
+        (Attribute.simplifiedAttribute pat)
+        $ case predF of
+            AndF     (And () t1 t2)       -> TermLike.mkAnd t1 t2
+            BottomF  _                    -> TermLike.mkBottom sort
+            CeilF    (Ceil () () t)       -> TermLike.mkCeil sort t
+            EqualsF  (Equals () () t1 t2) -> TermLike.mkEquals sort t1 t2
+            ExistsF  (Exists () v t)      -> TermLike.mkExists v t
+            FloorF   (Floor () () t)      -> TermLike.mkFloor sort t
+            ForallF  (Forall () v t)      -> TermLike.mkForall v t
+            IffF     (Iff () t1 t2)       -> TermLike.mkIff t1 t2
+            ImpliesF (Implies () t1 t2)   -> TermLike.mkImplies t1 t2
+            InF      (In () () t1 t2)     -> TermLike.mkIn sort t1 t2
+            NotF     (Not () t)           -> TermLike.mkNot t
+            OrF      (Or () t1 t2)        -> TermLike.mkOr t1 t2
+            TopF     _                    -> TermLike.mkTop sort
+
+fromPredicate_
+    :: InternalVariable variable
+    => Predicate variable
+    -> TermLike variable
+fromPredicate_ = fromPredicate predicateSort
 
-{-|'PredicateFalse' is a pattern for matching 'bottom' predicates.
--}
-pattern PredicateFalse :: Predicate variable
+data HasChanged = Changed | NotChanged
+    deriving (Show, Eq)
 
-{-|'PredicateTrue' is a pattern for matching 'top' predicates.
--}
-pattern PredicateTrue :: Predicate variable
+instance Semigroup HasChanged where
+    NotChanged <> x = x
+    Changed <> _ = Changed
 
-pattern PredicateFalse <- GenericPredicate (Recursive.project -> _ :< BottomF _)
-pattern PredicateTrue  <- GenericPredicate (Recursive.project -> _ :< TopF _)
+instance Monoid HasChanged where
+    mempty = NotChanged
 
-pattern PredicateAnd
-    :: Predicate variable -> Predicate variable -> Predicate variable
-pattern PredicateAnd p1 p2 <-
-    GenericPredicate (And_ _ (GenericPredicate -> p1) (GenericPredicate -> p2))
 {-|'isFalse' checks whether a predicate is obviously bottom.
 -}
-isFalse :: TopBottom patt => GenericPredicate patt -> Bool
+isFalse :: TopBottom patt => patt -> Bool
 isFalse = isBottom
 
-binaryOperator
-    :: (HasCallStack, InternalVariable variable)
-    => (TermLike variable -> TermLike  variable -> TermLike  variable)
-    -> Predicate variable -> Predicate variable -> Predicate variable
-binaryOperator operator (GenericPredicate first) (GenericPredicate second) =
-    GenericPredicate (operator first sortedSecond)
-  where
-    sortedSecond = TermLike.fullyOverrideSort (termLikeSort first) second
+isTrue :: TopBottom patt => patt -> Bool
+isTrue = not . isBottom
 
-{-| 'makeMultipleAndPredicate' combines a list of Predicates with 'and',
-doing some simplification.
--}
+makeTruePredicate' ::
+    InternalVariable variable
+    => (Predicate variable, HasChanged)
+makeTruePredicate' = (synthesize (TopF Top {topSort = ()})
+    , NotChanged)
```

Comment:
```suggestion
makeTruePredicate' =
    (synthesize (TopF Top {topSort = ()}), NotChanged)
```

Reply:
The reason why I wrote it like that is so that it is consistent in style with all the other maker functions: the body of the function followed by an extra, separate line where we specify whether the term changed or not. Should I still change it as in your example?

Reply:
All of the `make` functions in this module should change to follow our style guide.